regionGrowing.py

```diff
@@ -0,0 +1,192 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from volumeData import VolumeData
+from skimage.filters import threshold_yen
+from skimage.measure import label, regionprops
+from skimage.morphology import closing, square
+
+def srednia(arr, thresh):
+
+    # print('Tab: ', arr)
+    liczbaElementow = np.count_nonzero(arr >= thresh)
+    if liczbaElementow == 0:    # Nie dzielimy przez 0
+        return None
+
+    idxs = np.nonzero(arr >= thresh)
+    suma = sum(arr[idxs])
+    # print(f'Suma: {suma}, Liczba elementów: {liczbaElementow}')
+    return suma / liczbaElementow
+
+def sigma(arr, thresh):
+    """Odchylenie standardowe liczb powyżej progu"""
+    temp = arr - srednia(arr, thresh)
+    var = srednia((arr - srednia(arr,thresh))**2, thresh)
```

Comment:
1. Ponieważ będziesz potrzebował najczęściej i sredniej i sigma to powinna byc jedna funkcja ktora zwraca tuplę (mean, sigma) - wtedy nie liczysz sredniej 2x
2. Ta linia jest bledna bo zewnetrzna srednia nalozy progowanie na wartosci kwadratow odchylen an nie na wartosci macierzy

Proponowalbym najpierw wyznaczyc macierz po nalozeniu progu i zapisać jako MaskedArray: https://docs.scipy.org/doc/numpy/reference/maskedarray.generic.html#constructing-masked-arrays

Następnie użyć:
- https://docs.scipy.org/doc/numpy/reference/generated/numpy.ma.mean.html#numpy.ma.mean

Niestety numpy nie ma funkcji która będzie zwracać jednocześnie mean i std. Pozostaje coś w tym stylu:

```
m = np.ma.mean(a)
s = np.ma.mean((a-m)**2)
```

Reply:
A mógłbyś zajrzeć do najnowszej wersji tego algorytmu ? Jest on teraz metodą klasy VolumeData. Ponieważ próg, który przekazujemy do tego algorytmu miał służyć do wybrania woksela startowego. Kolejne woksele mają być dodawane do regionu, jeśli spełniają równanie: woksel - średnia <= c*sigma - bez względu na to, czy są większe niż próg czy mniejsze IMHO. W najnowszej wersji algorytmu średnią i sigmę liczę jako numpy.mean oraz sqrt(numpy.var).